src/pre_processing/extract_landscape.py

```diff
@@ -0,0 +1,165 @@
+        if only_test_data:
+            file_name = TEST_DATA_CACHED_PATH
+        elif only_train_data:
+            file_name = TRAIN_DATA_CACHED_PATH
+        else:
+            file_name = ALL_DATA_CACHED_PATH
+        logger.debug("Saving image_matrix to file {}".format(file_name))
+        np.save(file_name, image_matrix)
+
+    def _load_cached_image_matrix(only_train_data, only_test_data):
+        if only_test_data:
+            file_name = TEST_DATA_CACHED_PATH
+        elif only_train_data:
+            file_name = TRAIN_DATA_CACHED_PATH
+        else:
+            file_name = ALL_DATA_CACHED_PATH
+        logger.debug("Attempting to load cached file: {}".format(file_name))
+        return np.load(file_name)
+
+    if load_cached:
+        try:
+            image_matrix = _load_cached_image_matrix(only_train_data, only_test_data)
+            return image_matrix
+        except FileNotFoundError:
+            logger.debug("No file found, Processing data")
+
+    logger.debug("Getting facial vectors")
+    photos_matrix = _extract_photos_from_file(file_path, False, only_train_data, only_test_data)
+    facial_featutes_matrix = _create_correct_size_facial_matrix(only_train_data, only_test_data)
+    detector = dlib.get_frontal_face_detector()
+    predictor = dlib.shape_predictor(PREDICTOR_PATH)
+    faces_found = 0
+    faces_not_found = 0
+    logger.debug("Starting facial detection and feature extraction")
+    for i in range(len(photos_matrix)):
+        img = photos_matrix[i, :, :]
+        dets = detector(img, 2)
+        if len(dets) != 0:
+            facial_featutes_matrix[i, :, :] = _get_vectors_of_image_from_image(dets, predictor, img)
+            faces_found += 1
+        else:
+            facial_featutes_matrix[i, :, :] = np.zeros([68, 2])
+            faces_not_found += 1
+        if i % 5000 == 0:
+            logger.debug("Photos gone through {}".format(i))
+
+    logger.debug("Faces found: {}  Faces not found : {}".format(faces_found, faces_not_found))
+    _save_processed_image_matrix(facial_featutes_matrix, only_train_data, only_test_data)
+    return facial_featutes_matrix
+
+
+def _extract_photos_from_file(file_path=DATA_CSV_FILE, extract_first_only=False, only_train_data=False,
+                              only_test_data=False):
+    data_getter = GetDataFromCSV()
+    if only_train_data:
+        images_matrix, _ = data_getter.get_training_data()
+    elif only_test_data:
+        images_matrix, _ = data_getter.get_test_data()
+    elif extract_first_only:
+        images_matrix, _ = data_getter.get_first_record()
+    else:
+        images_matrix, _ = data_getter.get_all_data()
+    return images_matrix
+
+
+def _initialise_gui_window(img):
+    win = dlib.image_window()
+    win.clear_overlay()
+    win.set_image(img)
+    return win
+
+
+def _get_vectors_of_image_from_image(dets, predictor, img, win=None):
+    for k, d in enumerate(dets):
```

Comment:
this is the only thing left from my old review, where you mentioned that K is going to be the number of faces detected - i semantically don't understand that, as enumerate will return k as the index of the current dets